### ISSUE TITLE
Android: example build failed

### ISSUE THREAD

**Issue by CptFabulouso** [kmagiera/react-native-gesture-handler#102]
What I did:

- cloned repository
- ran `yarn install` in terminal in Example folder
- ran `react-native run-ios` -> this was ok
- ran `react-native run-android` -> failed

What went wrong:
```
Execution failed for task ':app:mergeDebugResources'.
> Error: java.lang.RuntimeException: com.android.builder.internal.aapt.AaptException: AAPT2 compile failed:
```
before the Error appears, there are few logs of this kind 
```
> Task :app:mergeDebugResources
/Users/Pavel/.gradle/caches/transforms-1/files-1.1/appcompat-v7-24.0.0.aar/74cbb38e018f31a50b2829379ad307ae/res/values/values.xml:235:5-90: AAPT: invalid dimen

```
so it may be releated to appcompat-v7-24.0.0?

my setup:
- running on macOS,
- trying to run on device Nexus 5X Android 8.1.0
installed versions of RN on system
- react-native-cli: 2.0.1
- react-native: 0.51.0

what I tried:
- Run `./gradlew clean`
- delete node_modules and run `npm install` instead of yarn
- installed android SDK 24.0.0 (then tried cleaning project again, restart terminal)
<img width="684" alt="snimek obrazovky 2018-01-12 v 12 11 44" src="https://user-images.githubusercontent.com/25827002/34872711-4affd1a2-f792-11e7-8128-a751e882dd6c.png">

**Comment by kmagiera** [comment]
Thanks for reporting @CptFabulouso 

I personally haven't encounter this issue and have seen many people building example app for Android without anyone reporting similar problems.

What I think could be happening is when you run `yarn install` from `Example` directory it ignores `files` section from `package.json` and installs all the files from gesture handler main directory including copy of `Example` directory which may contain some files that can impact the build. What I suggest is that you remove `node_modules`, do `yarn install` again (pls don't `npm install`), then delete `node_modules/react-native-gesture-handler/Example` and `node_modules/react-native-gesture-handler/node_modules` and try again.



If that doesn't help I suggest that you try the following:
 - In `Example/android/app/build.gradle` update `compileSdkVersion: 24`
 - OR try using 23.0.0 version of appcompat: `compile "com.android.support:appcompat-v7:23.0.0"`


Let me know if any of that steps help or maybe you managed to resolve it some other way.

**Comment by CptFabulouso** [comment]
the above suggestions did not help, however I've located the problem. It has something to do with float values in values.xml file generated during build and my locale setup. More in this [stackoverflow issue](https://stackoverflow.com/questions/44196672/aapt2-compile-failed-invalid-dimen-on-android-3-0-canary-1). Still I don't understand why it is happening, since I have the newest Android Studio version which should have this issue solved. 

The only thing that helped was to turn off the AAPT2 check by adding `android.enableAapt2=false` in android/gradle.properties. Not optimal, but I guess it's ok to do in example project.

I leave it to you if to close this issue, it does not seem to be directly related with the example project, still I did not encounter this issue elsewhere.

**Comment by kmagiera** [comment]
Nice finding @CptFabulouso !

I'm fine with adding that option to gradle properties if you'd like to send a PR.

There is also this new PR https://github.com/kmagiera/react-native-gesture-handler/pull/105 and I'm wondering if maybe that issue is resolved in the new version of build tools that PR is bumping.

**Comment by kmagiera** [comment]
@CptFabulouso  Would you be able to try build tools 27 and see if that fixes the problem?

**Comment by CptFabulouso** [comment]
you mean to update `buildToolsVersion '27.0.3'` in `Example/android/app/build.gradle` right? That did not solve the issue.
I also tried setting different versions of compileSdkVersion and appcompat, various combinations, but it has no effect, the error is always the same.

note that I did not clean the gradle between every combination, just few

**Comment by osdnk** [comment]
This issue seems to be inactive and thus I gonna close it.
If you still face the problem, please reopen it

**Comment by kmagiera** [comment]
Hey @CptFabulouso 

As I've recently encountered a very similar issue on a completely different project and not related to the use of this library I decided I'll post about it here, as this issue is what I've found when googling for the aapt2 problem.

So in my case the problem was my installation of Android Studio that upgraded android gradle plugin to a beta version that for some reason was lacking the aapt2 binaries. The only thing I had to change was the gradle file located under `android/build.gradle` that specifies which version of android gradle plugin should be used. I changed that to version 3.1.0 that worked ok for me (android studio upgraded to 3.2-beta or sth):
```
dependencies {
  classpath 'com.android.tools.build:gradle:3.1.0'
}
```
